### PR TITLE
Decoupled Constraints API object from internal representation

### PR DESCRIPTION
**Pull request by ellistarn**
**1. Issue, if available:**
https://github.com/aws/karpenter/issues/1350

**2. Description of changes:**
This is the first change in a series of changes to decouple the v1alpha5 API package from our internal node representation.

**3. How was this change tested?**

**4. Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/cloudprovider/aws/apis/v1alpha1/provider.go [hunk 1]
```diff
@@ -192,29 +186,29 @@ type BlockDevice struct {
...
-func Deserialize(constraints *v1alpha5.Constraints) (*Constraints, error) {
-	if constraints.Provider == nil {
+func Deserialize(provider *v1alpha5.Provider) (*Provider, error) {
+	if provider == nil {
 		return nil, fmt.Errorf("invariant violated: spec.provider is not defined. Is the defaulting webhook installed?")
 	}
-	aws := &AWS{}
-	_, gvk, err := Codec.UniversalDeserializer().Decode(constraints.Provider.Raw, nil, aws)
+	p := &Provider{}
+	_, gvk, err := Codec.UniversalDeserializer().Decode(provider.Raw, nil, p)
```
[the remark this answers is not in the document]
Comment (ellistarn) [reply pkg/cloudprovider/aws/apis/v1alpha1/provider.go:194]: To derisk, we can we cover this in another PR?

[the remark this answers is not in the document]
Comment (tzneal) [reply pkg/cloudprovider/aws/apis/v1alpha1/provider.go:194]: SGTM

File: pkg/cloudprovider/aws/cloudprovider.go [hunk 2]
```diff
@@ -101,45 +101,56 @@ func NewCloudProvider(ctx context.Context, options cloudprovider.Options) *Cloud
...
+	defaultLabels(provisioner)
+}
+
+func defaultLabels(provisioner *v1alpha5.Provisioner) {
+	for key, value := range map[string]string{
+		v1alpha5.LabelCapacityType: ec2.DefaultTargetCapacityTypeOnDemand,
+		v1.LabelArchStable:         v1alpha5.ArchitectureAmd64,
+	} {
+		if _, ok := provisioner.Spec.Labels[key]; ok {
+			return
```
Comment (bwagner5) [inline pkg/cloudprovider/aws/cloudprovider.go:144]: `continue` rather than `return` ?

File: pkg/cloudprovider/aws/cloudprovider.go [hunk 3]
```diff
@@ -101,45 +101,56 @@ func NewCloudProvider(ctx context.Context, options cloudprovider.Options) *Cloud
...
+		}
+		for _, requirement := range provisioner.Spec.Requirements.Requirements {
+			if requirement.Key == key {
+				return
```
Comment (bwagner5) [inline pkg/cloudprovider/aws/cloudprovider.go:148]: need to rework to go back around the outer for loop here too.

**Review by bwagner5 — CHANGES_REQUESTED** [review]

**Pull request merged**